builder/cloudstack/step_create_instance.go

```diff
@@ -65,7 +115,29 @@ func (s *stepCreateInstance) Run(state multistep.StateBag) multistep.StepAction
 	}
 
 	if config.UserData != "" {
-		ud, err := getUserData(config.UserData, config.HTTPGetOnly)
+		var httpPort uint = state.Get("http_port").(uint)
+		hostIp, err := hostIP()
+		if err != nil {
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+		common.SetHTTPIP(hostIp)
```

Comment:
I just tested removing setHTTPIP, and it still works, however it doesn't create the companion file to ```/tmp/packer-ca83bf7e-c2d2-3cef-8a7b-de12df2bff59-port```, which would be ```/tmp/packer-ca83bf7e-c2d2-3cef-8a7b-de12df2bff59-ip```. And I don't know what that file is used for.

So, all the other providers that use the http server sets the SetHTTPIP, but as previously mentioned, they could all have copied from the same source without understanding entirely what it's used for. Maybe @rickard-von-essen knows of what use that file is?